gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -999,13 +1011,17 @@ protected InputStream openStream(long bytesToRead) throws IOException {
           }
           break;
         } catch (IOException footerException) {
+          hasException = true;
```

Comment:
why not always increment?

Reply:
Seems like we throw exception only after maximum retries is reached so it made more sense to increment EXCEPTION_COUNT in that case only otherwise we will be incrementing at every retry which is not what EXCEPTION_COUNT is supposed to do. Please let me know if this can be done in better way.